pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -43,33 +47,29 @@ type fatalError string
 type transientError string
 
 func (e fatalError) Error() string {
-	return fmt.Sprintf("fatal error handling DeploymentConfig: %s", string(e))
+	return fmt.Sprintf("fatal error handling deploymentConfig: %s", string(e))
 }
 func (e transientError) Error() string {
-	return "transient error handling DeploymentConfig: " + string(e)
+	return "transient error handling deploymentConfig: " + string(e)
 }
 
 // Handle processes config and creates a new deployment if necessary.
 func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig) error {
+	// Inspect a deployment configuration every time the controller reconciles it
+	details, existingDeployments, latestDeploymentExists, err := c.findDetails(config)
+	if err != nil {
+		return err
+	}
+	c.updateDetails(config, details)
```

Comment:
Mentioned below- get the updated `config` and reassign to avoid resource conflicts on future updates.